homeassistant/components/touchlinesl/climate.py

```diff
@@ -0,0 +1,111 @@
+"""Roth Touchline SL climate integration implementation for Home Assistant."""
+
+from typing import Any
+
+from pytouchlinesl import Zone
+from pytouchlinesl.client.models import GlobalScheduleModel
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import TouchlineSLConfigEntry
+from .coordinator import TouchlineSLModuleCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: TouchlineSLConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Touchline devices."""
+    account = entry.runtime_data
+
+    for module in await account.modules():
+        coordinator = TouchlineSLModuleCoordinator(hass, module=module)
+        await coordinator.async_config_entry_first_refresh()
+        async_add_entities(
+            (
+                TouchlineSLZone(coordinator=coordinator, zone_id=z)
+                for z in coordinator.data["zones"]
+            ),
+            True,
+        )
+
+
+CONST_TEMP_PRESET_NAME = "Constant Temperature"
+
+
+class TouchlineSLZone(CoordinatorEntity, ClimateEntity):
+    """Roth Touchline SL Zone."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_mode = HVACMode.HEAT
+    _attr_hvac_modes = [HVACMode.HEAT]
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+
+    def __init__(self, *, coordinator, zone_id: int) -> None:
+        """Construct a Touchline SL climate zone."""
+        super().__init__(coordinator, context=zone_id)
+        self.id: int = zone_id
+        self._attr_unique_id = f"touchlinesl-zone-{self.id}"
```

Comment:
unique_id is unqiue per intergation per platform, so no need to prepend it with `touchlinesl-zone-`